anonymise/src/lib.rs

```diff
@@ -0,0 +1,328 @@
+
+    let output_tar_gz_path: PathBuf =
+        Path::new(&dir_output).join(Path::new(&tar_gz_file_name.replace("TDR", "TST")));
+    let uncompressed_folder_input_path: &PathBuf = &file.with_extension("").with_extension("");
+    let input_batch_reference: String = uncompressed_folder_input_path
+        .file_name()
+        .and_then(|name| name.to_str().map(|name| name.replace("TRE-", "")))
+        .ok_or(Error::new(
+            ErrorKind::InvalidInput,
+            "Cannot get a batch reference from the file name",
+        ))?;
+    let output_batch_reference: &String = &input_batch_reference.replace("TDR", "TST");
+
+    let extracted_output_original_name: PathBuf =
+        dir_output.join(PathBuf::from(&input_batch_reference));
+    let extracted_output_path: PathBuf = dir_output.join(PathBuf::from(output_batch_reference));
+
+    fs::create_dir_all(extracted_output_path.clone())?;
+
+    decompress_file(file, dir_output)?;
+
+    let metadata_input_file_path: &PathBuf = &extracted_output_path.join(PathBuf::from(format!(
+        "TRE-{input_batch_reference}-metadata.json"
+    )));
+    let metadata_output_file_path: &PathBuf = &extracted_output_path.join(PathBuf::from(format!(
+        "TRE-{output_batch_reference}-metadata.json"
+    )));
+
+    if extracted_output_path.exists() {
+        fs::remove_dir_all(&extracted_output_path)?;
+    }
+    fs::rename(extracted_output_original_name, &extracted_output_path)?;
+    fs::rename(metadata_input_file_path, metadata_output_file_path)?;
+
+    let mut metadata_json_value: Value = parse_metadata_json(metadata_output_file_path)?;
+
+    let docx_checksum =
+        create_docx_with_checksum(&extracted_output_path, &mut metadata_json_value)?;
+
+    update_json_file(
+        metadata_output_file_path,
+        docx_checksum,
+        &mut metadata_json_value,
+    )?;
+
+    if_present_delete(extracted_output_path.join(PathBuf::from(
+        format!("{}.xml", input_batch_reference).as_str(),
+    )))?;
+    if_present_delete(extracted_output_path.join(PathBuf::from("parser.log")))?;
+
+    tar_folder(
+        &output_tar_gz_path,
+        &extracted_output_path,
+        output_batch_reference,
+    )?;
+
+    let _ = fs::remove_dir_all(&extracted_output_path);
+    Ok(output_tar_gz_path)
+}
+
+/// # Creates a docx and returns a checksum
+///
+/// This creates a new docx file with the name parsed from the metadata filename.
+///
+/// It then writes the judgment name to the docx file and saves it to the output directory.
+/// If there is no judgment name, it uses the filename
+///
+/// Finally, it generates a sha256 checksum for the new file and returns it.
+fn create_docx_with_checksum(
+    extracted_output_path: &Path,
+    metadata_json_value: &mut Value,
+) -> Result<String, Error> {
+    let docx_file_name: &str = metadata_json_value["parameters"]["TRE"]["payload"]["filename"]
+        .as_str()
+        .ok_or("Filename is missing from the metadata json")
```

Comment:
Maybe just for clarity "'filename' is missing from the metadata.json" as it might seems as though the metadata.json's file name is missing?